Fail if `publicPath` is not specified

<!--
  ⚡️ katchow! We 💛 issues.

  Please - do not - remove this template.
  Please - do not - skip or remove parts of this template.
  Or your issue may be closed.

  👉🏽 Need help or support? Open a SUPPORT issue instead.
-->

- Webpack Version: 4.43.0
- Operating System (or Browser):
- Node Version:
- webpack-plugin-serve Version: 1.0.0

### How Do We Reproduce?

Remove `output.publicPath` in a webpack config, when using `webpack-plugin-serve@1.0.0`.

Problem was introduced in https://github.com/shellscape/webpack-plugin-serve/pull/156.

<!--
  Issues without minimal reproductions will be closed! Please provide one by:
  2. Work to isolate the problem and provide the exact steps in this issue, or
  3. Provide a minimal repository link (Read https://git.io/fNzHA for instructions). These may take more time to triage than the other options.
-->


### Expected Behavior

No error.


### Actual Behavior

Following error is thrown:

```
TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received type undefined
    at validateString (internal/validators.js:125:11)
    at join (path.js:1147:7)
    at WebpackPluginServe.hook (/Users/aleks/project/node_modules/webpack-plugin-serve/lib/index.js:194:22)
    at WebpackPluginServe.apply (/Users/aleks/project/node_modules/webpack-plugin-serve/lib/index.js:132:10)
    at webpack (/Users/aleks/project/node_modules/webpack/lib/webpack.js:51:13)
    at run (/Users/aleks/project/node_modules/webpack-nano/lib/compiler.js:16:20)
    at doeet (/Users/aleks/project/node_modules/webpack-nano/bin/wp.js:78:3)
    at process._tickCallback (internal/process/next_tick.js:68:7)
    at Function.Module.runMain (internal/modules/cjs/loader.js:834:11)
    at startup (internal/bootstrap/node.js:283:19)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
```

<!--
  The situation may arise where some small code snippets also need to be provided. In that situation, please add your code below using Fenced Code Blocks (https://help.github.com/articles/creating-and-highlighting-code-blocks/)
-->

--- a/lib/index.js
+++ b/lib/index.js
@@ -187,20 +187,22 @@ class WebpackPluginServe extends EventEmitter {
 
     // check static paths for publicPath. #100
     const { publicPath } = compiler.options.output;
-    let foundPath = false;
-    for (const path of this.options.static) {
-      const joined = join(path, publicPath);
-      if (existsSync(joined)) {
-        foundPath = true;
-        break;
+    if (publicPath) {
+      let foundPath = false;
+      for (const path of this.options.static) {
+        const joined = join(path, publicPath);
+        if (existsSync(joined)) {
+          foundPath = true;
+          break;
+        }
       }
-    }
 
-    /* istanbul ignore next */
-    if (!foundPath) {
-      this.log.warn(
-        chalk`{bold {yellow Warning}} The value of {yellow \`output.publicPath\`} was not found on the filesystem in any static paths specified\n`
-      );
+      /* istanbul ignore next */
+      if (!foundPath) {
+        this.log.warn(
+          chalk`{bold {yellow Warning}} The value of {yellow \`output.publicPath\`} was not found on the filesystem in any static paths specified\n`
+        );
+      }
     }
 
     // we do this emit because webpack caches and optimizes the hooks, so there's no way to detach

